src/main/java/net/minecraftforge/gametest/GameTestMain.java

```diff
@@ -0,0 +1,195 @@
+import net.minecraft.CrashReport;
+import net.minecraft.DefaultUncaughtExceptionHandler;
+import net.minecraft.SharedConstants;
+import net.minecraft.Util;
+import net.minecraft.commands.Commands;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.RegistryAccess;
+import net.minecraft.gametest.framework.GameTestBatch;
+import net.minecraft.gametest.framework.GameTestRegistry;
+import net.minecraft.gametest.framework.GameTestRunner;
+import net.minecraft.gametest.framework.GameTestServer;
+import net.minecraft.nbt.NbtOps;
+import net.minecraft.nbt.Tag;
+import net.minecraft.resources.RegistryReadOps;
+import net.minecraft.resources.RegistryWriteOps;
+import net.minecraft.server.Bootstrap;
+import net.minecraft.server.Eula;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.ServerResources;
+import net.minecraft.server.dedicated.DedicatedServerProperties;
+import net.minecraft.server.dedicated.DedicatedServerSettings;
+import net.minecraft.server.packs.PackType;
+import net.minecraft.server.packs.repository.FolderRepositorySource;
+import net.minecraft.server.packs.repository.PackRepository;
+import net.minecraft.server.packs.repository.PackSource;
+import net.minecraft.server.packs.repository.ServerPacksSource;
+import net.minecraft.world.level.DataPackConfig;
+import net.minecraft.world.level.GameRules;
+import net.minecraft.world.level.LevelSettings;
+import net.minecraft.world.level.levelgen.WorldGenSettings;
+import net.minecraft.world.level.storage.LevelResource;
+import net.minecraft.world.level.storage.LevelStorageSource;
+import net.minecraft.world.level.storage.LevelSummary;
+import net.minecraft.world.level.storage.PrimaryLevelData;
+import net.minecraft.world.level.storage.WorldData;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.File;
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+public class GameTestMain {
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    public static void main(String[] args) {
+        System.setProperty("forge.enablegametest", "true");
+        SharedConstants.tryDetectVersion();
+        OptionParser optionParser = new OptionParser();
+        OptionSpec<Void> helpOpt = optionParser.accepts("help").forHelp();
+        OptionSpec<Void> safeModeOpt = optionParser.accepts("safeMode", "Loads level with vanilla datapack only");
+        OptionSpec<String> savesDirOpt = optionParser.accepts("universe").withRequiredArg().defaultsTo(".");
+        OptionSpec<String> saveNameOpt = optionParser.accepts("world").withRequiredArg();
+        OptionSpec<BlockPos> spawnPosOpt = optionParser.accepts("spawnPos").withRequiredArg().withValuesConvertedBy(new BlockPosValueConverter()).defaultsTo(new BlockPos(0, 60, 0));
+        OptionSpec<Void> acceptEulaOpt = optionParser.accepts("acceptEula");
```

Comment:
Honestly this option strikes me as a bit odd given the fact that with how things are setup currently, this option is effectively a "require manually accepting the eula", and defaults to "false". So I am not really sure what benefit this option serves given I highly doubt someone would add it just to have to accept the eula as well when without it they could just run it without having to bother accepting it. Part of me feels like because of that this option should just be removed, or if not at the very least a description should be added that basically says this makes it require manually accepting it.

Reply:
I'm not sure what you mean here. If the option is enabled, the eula will be considered accepted, and the check on `eula.txt` will be skipped. If the option is not present, then `eula.txt` must have `eula=true` or the server will exist.

Reply:
On second look I had read the if statement that makes use of this backwards. Though what I had meant when I said with the description, I meant as the second parameter to `accepts` similar to how `safeMode` has, rather than just as a comment above it.